packages/runtime/src/message-type-contract.ts

```diff
@@ -22,8 +22,8 @@ export type PartialMessage<T extends object> = {
      T extends (Date | Uint8Array | bigint | boolean | string | number) ? T
    : T extends Array<infer U> ? Array<PartialField<U>>
    : T extends ReadonlyArray<infer U> ? ReadonlyArray<PartialField<U>>
-   : T extends { oneofKind: string } ? T
-   : T extends { oneofKind: undefined } ? T
+   : T extends { kind: string } ? T
+   : T extends { kind: undefined } ? T
```

Comment:
We want to capture all oneof groups here, and return `T`, because we don't want line 27 to apply to oneof groups.

We can target this a little bit better now with

```suggestion
   : T extends { kind: string; value: unknown } ? T
   : T extends { kind: undefined } ? T
```